[Update 10] Add GraphQL Changes to the Release Note (#9229)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md

- Introduced support for parsing XML with record types with default values as the expected type, using the default values where required (i.e., if a value corresponding to the record field is not present in the XML value).
- Introduced the option to choose between semantic and syntactic equality of XML elements and attributes.

#### `graphql` package

> - Added support for GraphQL query complexity analysis, which can be used to enhance the GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` record now includes a new field named `queryComplexityConfig` to configure the query complexity analysis.

Reviewer: ```suggestion
- Added support for GraphQL query complexity analysis, which can be used to enhance GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` annotation now includes a new field named `queryComplexityConfig` to configure query complexity analysis.
```

- Introduced support for parsing XML with record types with default values as the expected type, using the default values where required (i.e., if a value corresponding to the record field is not present in the XML value).
- Introduced the option to choose between semantic and syntactic equality of XML elements and attributes.

> #### `graphql` package

Reviewer: Shall we move this to after the data.yaml section to maintain alphabetic order?

Author: Updated.

        string|crypto:PublicKey certFile?;
    |};
    ```

> #### `graphql` package

Reviewer: This is not a backward-incompatible change, right? Currently listed under that section.

Author: Mistake. Fixed with 544bef54d20593b423108a1d5987f238f98ac864

> - Add support to GraphQL query complexity analysis, which can be used to enhance the GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` record now includes a new field `queryComplexityConfig` to configure the query complexity analysis.

Reviewer: ```suggestion
- Added support for GraphQL query complexity analysis, which can be used to enhance the GraphQL service security by mitigating the risk of denial-of-service attacks. With this update, the `graphql:ServiceConfig` record now includes a new field named `queryComplexityConfig` to configure query complexity analysis.
```

Author: Fixed with 544bef54d20593b423108a1d5987f238f98ac864

>     ```ballerina

Reviewer: Usually for library modules we seem to be adding just a one or two line description, without including examples. This is a bit of a deviation. https://github.com/ballerina-platform/ballerina-dev-website/blob/release-2201.10.0/downloads/swan-lake-release-notes/swan-lake-2201.10.0/RELEASE_NOTE.md#new-features-2

I am +1 for adding examples though.

Author: Removed the example to keep the consistency. We have to have a discussion on whether to add the examples here or not.

    import ballerina/graphql;

    @graphql:ServiceConfig {
        // Following are the default values, which can be overridden.
        queryComplexityConfig: {
            maxComplexity: 100,
            defaultFieldComplexity: 1,
            warnOnly: false
        }
    }
    service graphql:Service /graphql on new graphql:Listener(9090) {

>         // Defines a custom complexity value for the field

Reviewer: ```suggestion
        // Defines a custom complexity value for the field.
```

#### `xslt` package

> - Added parameter passing support for XSLT transformations.

Author: ```suggestion
- Added parameter passing support for XSLT transformations.

```
